### DISCUSSION TITLE
How can I specify context inside block from plugin?

### DISCUSSION THREAD

**Discussion by alisnic** [castwide/solargraph#518] [Q&A] [answered]
One of the things that I working on in https://github.com/alisnic/solar-rails is auto-complete story inside `config/routes.rb`. For that, cursor inside `Rails.application.routes.draw do` should evaluate to being in context of a instance `ActionDispatch::Routing::Mapper`:

```ruby
# config/routes.rb
Rails.application.routes.draw do
  # here all instance methods of ActionDispatch::Routing::Mapper should autocomplete
end
```

How can I achieve that?

**Comment by castwide** [comment, marked as answer]
Solargraph provides a custom `@yieldself` tag that lets you specify the context of a yielded block. Example:

```ruby
class Example
  def something
  end
end

# @yieldself [Example]
def run_in_example &block
  Example.new.instance_eval &block
end

run_in_example do
  something # <- Recognized as Example#something
end
```

The `@yieldself` tag should work inside an `@!override` as well.

**Reply by alisnic** [reply]
Thank you! Will try that

**Reply by alisnic** [reply]
> The @yieldself tag should work inside an @!override as well.

Like that? Does not seem to work for me 🤔 

```
# @!override ActionDispatch::Routing::RouteSet#draw
#   @yieldself [ActionDispatch::Routing::Mapper]
```

**Comment by castwide** [comment]
Does it work if you save the file and restart the editor?

**Reply by alisnic** [reply]
Works, had issues because of my refusal to put annotations in project folder, and instead bundle it in the gem 🙈 Thanks for speedy support!
